src/controller.c

```diff
@@ -46,94 +49,114 @@ int socket_connect(char *host, int portno) {
-
-  /* receive the response */
-  memset(response, 0, sizeof(response));
-  total = sizeof(response)-1;
-  received = 0;
-  do {
-      bytes = read(sockfd, response+received, total-received);
-      if (bytes < 0)
-          DebugMessage(M64MSG_ERROR, "ERROR reading response from socket");
-      if (bytes == 0)
-          break;
-      received+=bytes;
-  } while (received < total);
-
-  if (received == total)
-      DebugMessage(M64MSG_ERROR, "ERROR storing complete response from socket");
-
-  /* print the response */
-  #ifdef _DEBUG
-    DebugMessage(M64MSG_INFO, response);
-  #endif
-
-  /* parse the http response */
-  char * body = strtok(response, "\n");
-  for(int i=0; i<5; i++)
-    body = strtok(NULL,"\n");
-
-  /* parse the body of the response */
-  char * str;
-  int value;
-
-  body++; // skip first character. it is `[`
+    int sockfd = socket_connect(HOST, PORT);
 
-  /* x-axis */
-  str = strtok(body, ",");
-  value = atoi(str);
-  controller[0].buttons.X_AXIS = value;
-
-  /* y-axis */
-  str = strtok(NULL, ",");
-  value = atoi(str);
-  controller[0].buttons.Y_AXIS = value;
-
-  /* a button */
-  str = strtok(NULL, ",");
-  value = atoi(str);
-  controller[0].buttons.A_BUTTON = value;
-
-  /* b button */
-  str = strtok(NULL, ",");
-  value = atoi(str);
-  controller[0].buttons.B_BUTTON = value;
-
-  /* rb button */
-  str = strtok(NULL, ",]"); // include the close `]` as a token to just get the value
-  value = atoi(str);
-  controller[0].buttons.R_TRIG = value;
+    if (sockfd == -1) {
+        clear_controller();
+        return;
+    }
 
-  close(sockfd);
+    int bytes, sent, received, total;
+    char message[1024], response[4096]; // allocate more space than required.
+    sprintf(message, "GET / HTTP/1.0\r\n\r\n");
+
+    /* send the request */
+    total = strlen(message);
+    sent = 0;
+    do {
+        bytes = write(sockfd, message + sent, total - sent);
+        if (bytes < 0)
+            DebugMessage(M64MSG_ERROR, "ERROR writing message to socket");
+        if (bytes == 0)
+            break;
+        sent += bytes;
+    } while (sent < total);
+
+    /* receive the response */
+    memset(response, 0, sizeof(response));
+    total = sizeof(response) - 1;
+    received = 0;
+    do {
+        bytes = read(sockfd, response + received, total - received);
+        if (bytes < 0)
+            DebugMessage(M64MSG_ERROR, "ERROR reading response from socket");
+        if (bytes == 0)
+            break;
+        received += bytes;
+    } while (received < total);
+
+    if (received == total)
+        DebugMessage(M64MSG_ERROR, "ERROR storing complete response from socket");
+
+/* print the response */
+#ifdef _DEBUG
+    DebugMessage(M64MSG_INFO, response);
+#endif
+
+    /* parse the http response */
+    char *body = strtok(response, "\n");
+    for (int i = 0; i < 5; i++)
+        body = strtok(NULL, "\n");
+
+    /* parse the body of the response */
+	json_object *jsonObj = json_tokener_parse(body);
```

Comment:
You're right. Copy/paste grabbed a tab instead of spaces. My editor showed it aligned, but GitHub is rendering it differently... I'll fix it